mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ExecutorStateManagerImpl.java

```diff
@@ -63,62 +69,71 @@ public class ExecutorStateManagerImpl implements ExecutorStateManager {
         .build();
 
     @Override
-    public void putIfAbsent(TaskExecutorID taskExecutorID, TaskExecutorState state) {
+    public void trackIfAbsent(TaskExecutorID taskExecutorID, TaskExecutorState state) {
         this.taskExecutorStateMap.putIfAbsent(taskExecutorID, state);
         if (this.archivedState.getIfPresent(taskExecutorID) != null) {
             log.info("Reviving archived executor: {}", taskExecutorID);
             this.archivedState.invalidate(taskExecutorID);
         }
 
-        // add to buckets. however new executors won't have valid registration at this moment and requires marking
-        // again later when registration is ready.
+        tryMarkAvailable(taskExecutorID, state);
+    }
+
+    /**
+     * Add to buckets. however new executors won't have valid registration at this moment and requires marking
+     * again later when registration is ready.
+     * @param taskExecutorID taskExecutorID
+     * @param state state
+     * @return whether the target executor is marked as available.
+     */
+    public boolean tryMarkAvailable(TaskExecutorID taskExecutorID, TaskExecutorState state) {
         if (state.isAvailable() && state.getRegistration() != null) {
-            log.info("Marking executor {} as available for matching.", taskExecutorID);
+            TaskExecutorHolder teHolder = TaskExecutorHolder.of(taskExecutorID, state.getRegistration());
```

Comment:
I don't think we need this new DS. We should use the TaskExecutorState with a different comparator.

Reply:
TaskExecutorState has no ID in it. In the current state inside this ExecutorStateManagerImpl there are tracked on TE id -> its state separately and I think tracking ID + gen here should be enough and there is no strong benefit to merge states in.